include/zephyr/llext/symbol.h

```diff
@@ -155,33 +156,86 @@ struct llext_symtable {
 #endif
 /** @endcond */
 
+#if defined(LL_EXTENSION_BUILD)
+/* Extension build: No Kconfig conditionals */
+#define Z_EXPORT_SYMBOL_NAMED_IN_GROUP(group, sym_ident, sym_name)		\
+	Z_EXPORT_SYMBOL_NAMED(sym_ident, sym_name)
+#else
+/* LLEXT application: Export if relevant group is enabled */
+#define Z_EXPORT_SYMBOL_NAMED_IN_GROUP(group, sym_ident, sym_name)		\
+	IF_ENABLED(CONFIG_LLEXT_EXPORT_SYMBOL_GROUP_ ## group,			\
+		(Z_EXPORT_SYMBOL_NAMED(sym_ident, sym_name)))
```

Comment:
Actually, one remark I made in [this comment](https://github.com/zephyrproject-rtos/zephyr/pull/98890#issuecomment-3492343266):
```suggestion
	COND_CODE_1(CONFIG_LLEXT_EXPORT_SYMBOL_GROUP_ ## group,			\
		(Z_EXPORT_SYMBOL_NAMED(sym_ident, sym_name)),                   \
		(z_export_symbol_in_discarded_symbol_table(...))
```

But I won't block for this since it's more of a QoL feature - we can create an improvement issue to track this when PR is merged, and I'd be fine with implementing it myself (gives me a good excuse to get my hands dirty in LLEXT again... reminds me there's a cleanup I've been meaning to do since long but haven't gotten around to 🙂).